concrete-core/src/backends/core/implementation/engines/mod.rs

```diff
@@ -32,45 +35,38 @@ impl Display for CoreError {
 
 impl Error for CoreError {}
 
-use crate::backends::core::private::crypto::bootstrap::FourierBskBuffers;
-use crate::prelude::{FourierLweBootstrapKey32, FourierLweBootstrapKey64, LweBootstrapKeyEntity};
-use concrete_commons::parameters::{GlweSize, PolynomialSize};
-use std::collections::BTreeMap;
-
 #[derive(Ord, PartialOrd, Eq, PartialEq)]
 pub(crate) struct FourierBufferKey(pub PolynomialSize, pub GlweSize);
 
 /// The main engine exposed by the core backend.
 pub struct CoreEngine {
     secret_generator: ImplSecretRandomGenerator,
     encryption_generator: ImplEncryptionRandomGenerator,
-    fourier_bsk_buffers_u32: BTreeMap<FourierBufferKey, FourierBskBuffers<u32>>,
-    fourier_bsk_buffers_u64: BTreeMap<FourierBufferKey, FourierBskBuffers<u64>>,
+    fourier_bsk_buffers_u32: BTreeMap<FourierBufferKey, FourierBuffers<u32>>,
```

Comment:
In the first place with use a map because we need different sizes of buffers (for different poly size and glwe dims) to be stored in the engine. I think we used a BTreeMap (@tmontaigu ?) because it has a faster lookup than others. That being said, we should ensure that the lookup is faster than allocating the buffers (I actually have no ideas about the relative timings of both operations :sweat_smile: ). I intend perform benchmarks on that. If the allocation is faster, maybe we can find a way to have a faster lookup, or we can come back to allocating buffers every times.

Reply:
300 issues! 🎉